packages/runtime/src/runtime/messaging/subservices/lifetime.ts

```diff
@@ -0,0 +1,71 @@
+import { MessagingClient, uuid } from '@botpress/messaging-client'
+import { BotConfig } from 'botpress/runtime-sdk'
+import { CloudMessagingChannel } from '../../cloud/messaging'
+import { ConfigProvider } from '../../config'
+import { MessagingInteractor } from './interactor'
+
+export class MessagingLifetime {
+  private botIdToClientId: { [botId: string]: uuid } = {}
+  private clientIdToBotId: { [clientId: uuid]: string } = {}
+  private httpClients: { [botId: string]: MessagingClient } = {}
+
+  constructor(private configProvider: ConfigProvider, private interactor: MessagingInteractor) {}
+
+  getClientId(botId: string) {
+    return this.botIdToClientId[botId]
+  }
+
+  getBotId(clientId: uuid) {
+    return this.clientIdToBotId[clientId]
+  }
+
+  getHttpClient(botId: string): MessagingClient {
+    return this.httpClients[botId]
+  }
+
+  async loadMessagingForBot(botId: string) {
+    const config = await this.configProvider.getBotConfig(botId)
+
+    this.loadClientId(botId, config)
+
+    if (process.OAUTH_ENDPOINT) {
+      this.loadOAuth(botId, config)
+    }
+  }
+  private loadClientId(botId: string, config: BotConfig) {
+    if (!config.messaging) {
+      throw new Error(`Bot ${botId} does not have a messaging config`)
+    }
+
+    const { clientId, clientToken, webhookToken } = config.messaging
+    this.clientIdToBotId[clientId] = botId
+    this.botIdToClientId[botId] = clientId
+
+    this.httpClients[botId] = this.interactor.createHttpClientForBot(clientId, clientToken, webhookToken)
+    this.interactor.client.start(clientId, { clientToken, webhookToken })
+  }
+  private loadOAuth(botId: string, config: BotConfig) {
+    const cloud = config.cloud
+    if (!cloud) {
+      throw new Error(`Bot ${botId} does not have a cloud config`)
```

Comment:
I suggest you use `VError` when possible, it brings many advantages over the built-in `Error` type.
We can trust this package, it's been written by the creators of Node JS (they know their stuff)